### PR TITLE
Make rsx::get_address report caller location

### PR DESCRIPTION
**Pull request by elad335**
Report line and filename of caller rsx::get_address upon failure of address translation.

Note: Also added verification for RSXIO REPORT memory for offsets out of range: offset must be lower than 10m.

### CODE REVIEW

**Comment by kd-11** [comment]
Code is fine and logic is sound (and needed to aid debugging) but once again https://github.com/RPCS3/rpcs3/issues/7220 stands in the way. And no, its not about ease of fixing or correctness, think of it as a fence so that any bug reports after it will refer to builds created after it is resolved. That is the whole reasoning behind a code freeze.

**Comment by tsou21** [comment]
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/4140822/RPCS3.log.gz)
[TTY.log](https://github.com/RPCS3/rpcs3/files/4140824/TTY.log)

**Comment by elad335** [comment]
Use the pull request build, I don't need a log of master build.

**Comment by elad335** [comment]
RSX no longer in maintaince mode merge it. 🎉

File: rpcs3/Emu/RSX/RSXThread.cpp [hunk 1]
```diff
@@ -836,8 +847,8 @@ namespace rsx
 		const rsx::index_array_type type = rsx::method_registers.index_type();
 		const u32 type_size = get_index_type_size(type);
 
-		u32 address = rsx::get_address(rsx::method_registers.index_array_address(), rsx::method_registers.index_array_location());
-		address &= ~(type_size - 1); // Force aligned indices as realhw
+		// Force aligned indices as realhw
+		const u32 address = (0 - type_size) & get_address(rsx::method_registers.index_array_address(), rsx::method_registers.index_array_location(), HERE);
```
Comment (kd-11) [inline rpcs3/Emu/RSX/RSXThread.cpp]: I can't remember the hw test for this. Was is tested?

Comment (elad335) [reply rpcs3/Emu/RSX/RSXThread.cpp]: yes.

File: rpcs3/Emu/Cell/Modules/cellGcmSys.cpp [hunk 2]
```diff
@@ -142,7 +142,7 @@ vm::ptr<CellGcmReportData> cellGcmGetReportDataAddressLocation(u32 index, u32 lo
 		cellGcmSys.error("cellGcmGetReportDataAddressLocation: Wrong local index (%d)", index);
 	}
 
-	return vm::ptr<CellGcmReportData>::make(g_fxo->get<gcm_config>()->gcm_info.label_addr + 0x1400 + index * 0x10);
+	return vm::ptr<CellGcmReportData>::make(g_fxo->get<gcm_config>()->gcm_info.label_addr + ::offset32(&RsxReports::report) + index * 0x10);
```
Comment (kd-11) [inline rpcs3/Emu/Cell/Modules/cellGcmSys.cpp]: Obviously the cellGcmSys changes do not belong in this commit.

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellGcmSys.cpp]: hmm its a clean up for similar use case in rsx::get_address, so might as well clean it all.

Comment (kd-11) [reply rpcs3/Emu/Cell/Modules/cellGcmSys.cpp]: My complaint is because its a completely different module (fw) vs the other stuff which is unrelated and not a cleanup. get_address change is not cosmetic, it is functional and adds logging. Changes to gcmsys are doing something different, it would have been better to split. It actually makes more sense to merge the changes to gcmsys with the commit that follows this one and is very easy to do with a soft reset and file reselection.

**Comment by elad335** [comment]
All review comments solved.
